### DISCUSSION TITLE
👩‍💻 Support for non-persisted communication (Ie. Signals)

### DISCUSSION THREAD

**Discussion by skylerjokiel** [microsoft/FluidFramework#12345] [Azure Features & Functionality]
|||
|--------------|-----------|
| Status                 | 👩‍💻 In Progress |
| Target Delivery  | CY23 Q1           | 

## Problem
Currently the only data structures available to developers are persisted ones (Eg. SharedMap/SharedString). Signals are a way for developers to communicate with other connected clients within the session without incurring the cost of storage associated with persisted data structures. 

## Goal
Developers using AzureClient and Azure Fluid Relay can leverage non-persisted communication.

## Use Cases
The primary scenario is inner session communication of metadata. (Eg. user presence)

**Comment by danmarshall** [comment]
I'm also considering this scenario, and if this type of session metadata (low latency, no persistence) should be part of the DDS or not. I'm leaning towards it being a separate service since these signals have a different nature.

**Reply by danmarshall** [reply]
For example, using [Azure PubSub](https://azure.microsoft.com/en-us/products/web-pubsub/#overview) alongside with Fluid Relay.
